instrumentation/instacosmos/client.go

```diff
@@ -0,0 +1,126 @@
+// (c) Copyright IBM Corp. 2023
+
+//go:build go1.18
+// +build go1.18
+
+package instacosmos
+
+import (
+	"context"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime"
+	"github.com/Azure/azure-sdk-for-go/sdk/data/azcosmos"
+	instana "github.com/instana/go-sensor"
+)
+
+const emptyPrimaryKey string = ""
+
+// Client is the interface that wraps the methods of *azcosmos.Client
+type Client interface {
+	Endpoint() string
+	NewContainer(collector instana.TracerLogger, databaseID string, containerID string) (ContainerClient, error)
```

Comment:
I think we could make the API leaner for users.
Since we have `instaClient`, which embeds `azcosmos.Client`, maybe we don't need the duplicated `NewDatabase()` and `NewContainer()` methods (our version of them that requires an instance of `instana.TracerLogger`).
Instead, we could have the instance of `instana.TracerLogger` as a private property of instaClient, which would be accessible by `NewDatabase()` and `NewContainer()`. This way you wouldn't need to pass `collector` as an argument, and instead would "overwrite" the original methods from azcosmos, eliminating then the duplicated method implementations.

Users would also benefit from it, as they would have to change less existing code to use our API.

I hope this makes sense. Maybe there is some limitation in the Cosmos SDK that prevents us to make this change. But since we have the Client interface and the instaClient implementation, it should be fine to create the instance of `instana.TracerLogger` only when the client is created.

Reply:
> I think we could make the API leaner for users. Since we have `instaClient`, which embeds ...

An example of what I am trying to explain is the instrumentation of redigo:

Here you can see sensor is part of the struct:
https://github.com/instana/go-sensor/blob/main/instrumentation/instaredigo/instaredigo.go#L24

Here you see the sensor being sent to a new instance of the struct (among other properties):
https://github.com/instana/go-sensor/blob/main/instrumentation/instaredigo/instaredigo.go#L46

Here you see the sensor from the struct being used. Note that `Send` does not require a reference to the sensor, as both are properties of the struct :)
https://github.com/instana/go-sensor/blob/main/instrumentation/instaredigo/instaredigo.go#L187

Reply:
It makes sense to me as well. Good Suggestion. Updated the code with same